src/pipeline.js

```diff
@@ -169,11 +171,42 @@ class Pipeline {
    * @returns {Pipeline} this
    */
   once(f) {
+    Pipeline.describe(f);
     this._oncef = f;
     this._last = null;
     return this;
   }
 
+  /**
+   * This helper method generates a human readable name for a given function
+   * It will include:
+   * - the name of the function or "anonymous"
+   * - the name of the function that called describe
+   * - the name and code location of the function that called the function before
+   * @param {Function} f
+   */
+  static describe(f) {
+    if (f.alias) {
+      return f.alias;
+    }
+    if (!f.alias && f.name) {
+      // eslint-disable-next-line no-param-reassign
+      f.alias = f.name;
+    } else if (!f.name && !f.alias) {
+      // eslint-disable-next-line no-param-reassign
+      f.alias = 'anonymous';
+    }
+
+    Error.captureStackTrace(f);
+    const [, current, injector, caller] = f.stack.split('\n').map(s => s.replace(/^.*at /, ''));
```

Comment:
does this work in all locales? maybe better to use https://www.npmjs.com/package/traceback

Reply:
Good point.